lib/hooks/useTheme.tsx

```diff
@@ -7,19 +8,29 @@ export default function useTheme(theme?: Maybe<string>) {
   const [isReady, setIsReady] = useState(false);
   useEffect(() => {
     setIsReady(true);
-    if (!theme) {
-      return;
-    }
-    if (!theme && window.matchMedia("(prefers-color-scheme: dark)").matches) {
-      document.documentElement.classList.add("dark");
-    } else if (!theme) {
-      /** Uncovered case */
-    } else {
-      document.documentElement.classList.add(theme);
-    }
   }, []);
+  function Theme() {
+    const themeString = theme ? `"${theme}"` : null;
+    return (
+      <Head>
+        <script
+          dangerouslySetInnerHTML={{
+            __html: `(function (theme) {
+            if (!theme && window.matchMedia("(prefers-color-scheme: dark)").matches) {
+              document.documentElement.classList.add("dark");
+            } else if (!theme) {
+              /** Uncovered case */
+            } else {
+              document.documentElement.classList.add(theme);
+            }
+          })(${themeString})`,
+          }}></script>
```

Comment:
I can see that this works only on refresh. Can we make this work so we listen for that media query instead and changes display live?